keyboards/pearlboards/pearl/rev1/keymaps/cijanzen/keymap.c

```diff
@@ -0,0 +1,77 @@
+                   ____,      ____,      ____,      KC_DEL,    KC_DEL,               ____,                 KC_RGUI,   ____
+    ),
+    // FROW LAYER AND ARROWS (Fn2)
+    [2] = LAYOUT_all(
+        KC_ESC,    KC_F1,     KC_F2,     KC_F3,     KC_F4,     KC_F5,     KC_F6,     KC_F7,     KC_F8,     KC_F9,     KC_F10,    KC_F11,     KC_F12,
+        ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      KC_PGUP,   ____,                  ____,
+        ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      KC_HOME,   KC_PGDN,   KC_END,                ____,
+                   MO(4),     ____,      ____,      ____,      ____,                 ____,                 ____,      ____
+    ),
+    // MEDIA AND RGB (Fn3)
+    [3] = LAYOUT_all(
+        ____,      RGB_M_P,   RGB_M_B,   RGB_M_R,   RGB_M_SW,  RGB_M_SN,  RGB_M_K,   RGB_M_X,   RGB_M_G,   KC_MPRV,   KC_MPLY,   KC_MNXT,    KC_DEL,
+        ____,      ____,      ____,      RGB_HUI,   RGB_SAI,   RGB_VAI,   ____,      ____,      ____,      ____,      ____,                  ____,
+        ____,      RGB_MOD,   RGB_TOG,   RGB_HUD,   RGB_SAD,   RGB_VAD,   BL_TOGG,   BL_STEP,   ____,      ____,      ____,                  ____,
+                   ____,      ____,      ____,      ____,      ____,                 ____,                 ____,      ____
+    ),
+    // UTIL (Fn1+Fn3)
+    [4] = LAYOUT_all(
+        KC_SLP,    ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      KC_PSCR,    RESET,
+        ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,                  ____,
+        ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,      ____,                  ____,
+                   ____,      ____,      ____,      ____,      ____,                 ____,                 ____,      ____
+    ),
+};
+
+/**
+ * Status LED layer indicators courtesy of jetpacktuxedo's firmware
+ */
+uint32_t layer_state_set_kb(uint32_t state)
+{
+  if (state & (1<<1)) { // if we are on layer 1
+    PORTD |= (1 << PD0); // light num lock led
+  } else if (state & (1<<2)) { // if we are on layer 2
+
+    if (state & (1<<4)) {
+      PORTD |= (1 << PD0); // light all indicator leds
+      PORTD |= (1 << PD1);
+      PORTD |= (1 << PD6);
+    } else {
+      PORTD &= ~(1 << PD0);
+      PORTD |= (1 << PD1); // light caps lock led
+      PORTD &= ~(1 << PD6);
+    }
+  } else if (state & (1<<3)) { // if we are on layer 3
+    PORTD |= (1 << PD6); // light scroll lock led
+  } else if (state & (1<<4)) { // if we are stuck on layer 4
+    PORTD |= (1 << PD0); // light all indicator leds
+    PORTD |= (1 << PD1);
+    PORTD |= (1 << PD6);
+  } else {
+    PORTD &= ~(1 << PD0);
+    PORTD &= ~(1 << PD1);
+    PORTD &= ~(1 << PD6);
+  }
+
+  return state;
+}
```

Comment:
This needs a lot of ... work. 

uint32_t should NOT be used here for the layer mask, but instead should use `layer_state_t`.

user level function should be used here, not the kb level one (eg, `layer_state_set_user`).

`layer_state_cmp(state, x)` can/should be used for checking if a layer is on. 

GPIO commands should be used here, not AVR specific commands.
https://docs.qmk.fm/#/internals_gpio_control

Reply:
This part is probably my fault, but in my defense I wrote it before `layer_state_cmp` existed (and possibly before `layer_state_t` existed as well? This is like 5 year old code being yoinked through time.

Reply:
@Koobaczech , I can take a whack at modernizing this sometime this week if you'd like, but if you're feeling brave enough to take a swing at it you can look at the very similar code in [my prime_e keymap](https://github.com/qmk/qmk_firmware/blob/master/keyboards/primekb/prime_e/keymaps/jetpacktuxedo/keymap.c#L84) that I think is written the way @drashna is suggesting.

My pearl is buried in a mountain of boards right now so it might be a few days before I can dig it out and test on hardware 😅

Reply:
@jetpacktuxedo Oh boy haha. I would LOVE your help and input man, if its not too much to ask please. I definitely understand his changes but I do not have any old hardware around anymore x! Ty man, if you want any help too just lmk!